ufo2fontir/src/source.rs

```diff
@@ -140,10 +140,20 @@ impl DesignSpaceIrSource {
     ) -> Result<StateSet, Error> {
         let mut font_info = StateSet::new();
         font_info.track_file(&self.designspace_file)?;
+        let default_master = default_master(designspace)
+            .ok_or_else(|| Error::NoDefaultMaster(self.designspace_file.clone()))?;
+
         for source in designspace.sources.iter() {
             let ufo_dir = self.designspace_dir.join(&source.filename);
-            for filename in ["fontinfo.plist", "layercontents.plist"] {
+            for filename in ["fontinfo.plist", "layercontents.plist", "lib.plist"] {
+                // Only track lib.plist for the default master
+                if filename == "lib.plist" && source != default_master {
+                    continue;
+                }
+
                 let font_info_file = ufo_dir.join(filename);
+                // TODO: this is incorrect; several of these files are optional
+                // File tracking curently assumes you only track extant files so keep it for now
                 if !font_info_file.is_file() {
```

Comment:
only layercontents.plist is required for UFOv3, lib.plist and fontinfo.plist are optional, though the latter is most likely present as you can't do much, let alone compile a font, if you're not at least given some basic parameters like UPEM and a few others.

Reply:
Tricky bit is if one appears we would want to trigger a recompile. Tracking in https://github.com/googlefonts/fontmake-rs/issues/49.